apis/nodetopology/v1/types.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=tp,scope=Cluster
+// +kubebuilder:storageversion
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeTopology describes the current subnets used by this GKE cluster and its
+// current node zone information.
+// A NodeTopology will be created one per GKE cluster.
+type NodeTopology struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NodeTopologySpec   `json:"spec,omitempty"`
+	Status NodeTopologyStatus `json:"status,omitempty"`
+}
+
+// NodeTopologySpec is the spec for a NodeTopology resource
+type NodeTopologySpec struct{}
```

Comment:
Will we ever have a Spec here? Is this a status only resource?

Reply:
Yes that's a status only resource. This is intended to be a placeholder, just like what we did for NEG CRD https://github.com/kubernetes/ingress-gce/blob/master/pkg/apis/svcneg/v1beta1/types.go#L39.